package/upgrade/lib.sh

```diff
@@ -123,15 +123,20 @@ ${CURRENT_VERSION}"
   fi
 }
 
-wait_repo()
+get_repo_vm_status()
 {
-  local repo_vm_status
+  kubectl get virtualmachines.kubevirt.io $UPGRADE_REPO_VM_NAME -n $UPGRADE_NAMESPACE -o=jsonpath='{.status.printableStatus}'
+}
 
+wait_repo()
+{
   # Start upgrade repo VM in case it's shut down due to migration timeout or job failure
-  repo_vm_status=$(kubectl get virtualmachines.kubevirt.io $UPGRADE_REPO_VM_NAME -n $UPGRADE_NAMESPACE -o=jsonpath='{.status.printableStatus}')
-  if [ "$repo_vm_status" != "Running" ]; then
+  until [[ "$(get_repo_vm_status)" == "Running" ]]
+  do
+    echo "Try to bring up the upgrade repo VM..."
```

Comment:
VM may take time to startup, here as long as it is not in "Running", `virtctl start` will run every 10s.

Please double check, if vm is staring, the multi-time `virtctl start` will hurt or not.

Or only 'start' when vm is stopped or error ?

Reply:
There's no harm executing `virtctl start` multiple times. Update the PR to the original version. Please check again, thank you!